<ins>Instructions</ins>:
1. Apply logistic regression on the Bank Personal Loan dataset  
   {DV: personal loan / IDV: all other features except ID & ZIPCode}<br><br>

2. Apply logistic regression on the attrition project (use LabelEncoder)<br><br>

3. Apply linear regression on the Real Estate dataset — find R².  
   Test 5 models: {price/sqft_living}, {price/bedrooms}, {price/bathrooms}, {price/floors}, {price/sqft_living,bedrooms,bathrooms,floors}

Explain each results: what do we infer from it? ie. how much impact has the square feet feature on the sale's price

In [1]:
# Files: https://github.com/LetsUpgrade/AI-ML-July-2020/tree/master/Day-21/dataset
import pandas as pd
import matplotlib.pyplot as plt

## 1. Bank Personal Loan

### Import dataset

In [2]:
df = pd.read_excel('Bank_Personal_Loan_Modelling.xlsx', sheet_name=1)
df.head()

ID  Age  Experience  Income  ZIP Code  Family  CCAvg  Education  Mortgage  \
0   1   25           1      49     91107       4    1.6          1         0   
1   2   45          19      34     90089       3    1.5          1         0   
2   3   39          15      11     94720       1    1.0          1         0   
3   4   35           9     100     94112       1    2.7          2         0   
4   5   35           8      45     91330       4    1.0          2         0   

   Personal Loan  Securities Account  CD Account  Online  CreditCard  
0              0                   1           0       0           0  
1              0                   1           0       0           0  
2              0                   0           0       0           0  
3              0                   0           0       0           0  
4              0                   0           0       0           1

In [3]:
df.isna().sum()

ID                    0
Age                   0
Experience            0
Income                0
ZIP Code              0
Family                0
CCAvg                 0
Education             0
Mortgage              0
Personal Loan         0
Securities Account    0
CD Account            0
Online                0
CreditCard            0
dtype: int64

### Split dataset

In [4]:
Y = df['Personal Loan']
X = df.drop(['Personal Loan', 'ID', 'ZIP Code'], axis=1)

X.head()

Age  Experience  Income  Family  CCAvg  Education  Mortgage  \
0   25           1      49       4    1.6          1         0   
1   45          19      34       3    1.5          1         0   
2   39          15      11       1    1.0          1         0   
3   35           9     100       1    2.7          2         0   
4   35           8      45       4    1.0          2         0   

   Securities Account  CD Account  Online  CreditCard  
0                   1           0       0           0  
1                   1           0       0           0  
2                   0           0       0           0  
3                   0           0       0           0  
4                   0           0       0           1

### Train model

In [5]:
import statsmodels.api as sm

# Add constant to dataset
X1 = pd.DataFrame(
    sm.add_constant(X.values),
    columns=['const']+list(X.columns))
X1.head()

const   Age  Experience  Income  Family  CCAvg  Education  Mortgage  \
0    1.0  25.0         1.0    49.0     4.0    1.6        1.0       0.0   
1    1.0  45.0        19.0    34.0     3.0    1.5        1.0       0.0   
2    1.0  39.0        15.0    11.0     1.0    1.0        1.0       0.0   
3    1.0  35.0         9.0   100.0     1.0    2.7        2.0       0.0   
4    1.0  35.0         8.0    45.0     4.0    1.0        2.0       0.0   

   Securities Account  CD Account  Online  CreditCard  
0                 1.0         0.0     0.0         0.0  
1                 1.0         0.0     0.0         0.0  
2                 0.0         0.0     0.0         0.0  
3                 0.0         0.0     0.0         0.0  
4                 0.0         0.0     0.0         1.0

In [6]:
# Train
model  = sm.Logit(Y, X1)
result = model.fit()
result.summary()

Optimization terminated successfully.
         Current function value: 0.128435
         Iterations 9


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:          Personal Loan   No. Observations:                 5000
Model:                          Logit   Df Residuals:                     4988
Method:                           MLE   Df Model:                           11
Date:                Thu, 06 Aug 2020   Pseudo R-squ.:                  0.5938
Time:                        08:13:26   Log-Likelihood:                -642.18
converged:                       True   LL-Null:                       -1581.0
Covariance Type:            nonrobust   LLR p-value:                     0.000
======================================================================================
                         coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------
const                -12.1928      1.645     -7.411      0.000     -15.417      -8.968
Age                   -0.0536      0.061     -0.874      0.382      -0.174       0.067
Experience             0.0638      0.061      1.046      0.295      -0.056       0.183
Income                 0.0546      0.003     20.831      0.000       0.049       0.060
Family                 0.6958      0.074      9.364      0.000       0.550       0.841
CCAvg                  0.1240      0.040      3.127      0.002       0.046       0.202
Education              1.7362      0.115     15.088      0.000       1.511       1.962
Mortgage               0.0005      0.001      0.856      0.392      -0.001       0.002
Securities Account    -0.9368      0.286     -3.277      0.001      -1.497      -0.377
CD Account             3.8225      0.324     11.800      0.000       3.188       4.457
Online                -0.6752      0.157     -4.298      0.000      -0.983      -0.367
CreditCard            -1.1197      0.205     -5.462      0.000      -1.522      -0.718
======================================================================================
"""

Not all variables are important: Age, Experience and Mortgage are not needed to predict Personal Loan

## 2. Attrition

### Import dataset

In [7]:
df = pd.read_csv('general_data.csv')
df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

### Handle missing value

In [8]:
df.isna().sum()

Age                         0
Attrition                   0
BusinessTravel              0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeID                  0
Gender                      0
JobLevel                    0
JobRole                     0
MaritalStatus               0
MonthlyIncome               0
NumCompaniesWorked         19
Over18                      0
PercentSalaryHike           0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           9
TrainingTimesLastYear       0
YearsAtCompany              0
YearsSinceLastPromotion     0
YearsWithCurrManager        0
dtype: int64

In [9]:
df.dropna(inplace=True)
df.isna().sum()

Age                        0
Attrition                  0
BusinessTravel             0
Department                 0
DistanceFromHome           0
Education                  0
EducationField             0
EmployeeCount              0
EmployeeID                 0
Gender                     0
JobLevel                   0
JobRole                    0
MaritalStatus              0
MonthlyIncome              0
NumCompaniesWorked         0
Over18                     0
PercentSalaryHike          0
StandardHours              0
StockOptionLevel           0
TotalWorkingYears          0
TrainingTimesLastYear      0
YearsAtCompany             0
YearsSinceLastPromotion    0
YearsWithCurrManager       0
dtype: int64

### Handle categorical data

In [10]:
df.dtypes

Age                          int64
Attrition                   object
BusinessTravel              object
Department                  object
DistanceFromHome             int64
Education                    int64
EducationField              object
EmployeeCount                int64
EmployeeID                   int64
Gender                      object
JobLevel                     int64
JobRole                     object
MaritalStatus               object
MonthlyIncome                int64
NumCompaniesWorked         float64
Over18                      object
PercentSalaryHike            int64
StandardHours                int64
StockOptionLevel             int64
TotalWorkingYears          float64
TrainingTimesLastYear        int64
YearsAtCompany               int64
YearsSinceLastPromotion      int64
YearsWithCurrManager         int64
dtype: object

In [11]:
df.drop(['EmployeeID', 'EmployeeCount', 'Over18', 'StandardHours'], axis=1, inplace=True)
df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  Gender  JobLevel                    JobRole  \
0          2  Life Sciences  Female         1  Healthcare Representative   
1          1  Life Sciences  Female         1         Research Scientist   
2          4          Other    Male         4            Sales Executive   
3          5  Life Sciences    Male         3            Human Resources   
4          1        Medical    Male         1            Sales Executive   

  MaritalStatus  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
0       Married         131160                 1.0                 11   
1        Single          41890                 0.0                 23   
2       Married         193280                 1.0                 15   
3       Married          83210                 3.0                 11   
4        Single          23420                 4.0                 12   

   StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0                1.0                      6               1   
1                 1                6.0                      3               5   
2                 3                5.0                      2               5   
3                 3               13.0                      5               8   
4                 2                9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4

In [12]:
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder()

for col in df.select_dtypes('object').columns:
    df[col] = encoder.fit_transform(df[col])

df.head()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  Education  \
0   51          0               2           2                 6          2   
1   31          1               1           1                10          1   
2   32          0               1           1                17          4   
3   38          0               0           1                 2          5   
4   32          0               2           1                10          1   

   EducationField  Gender  JobLevel  JobRole  MaritalStatus  MonthlyIncome  \
0               1       0         1        0              1         131160   
1               1       0         1        6              2          41890   
2               4       1         4        7              1         193280   
3               1       1         3        1              1          83210   
4               3       1         1        7              2          23420   

   NumCompaniesWorked  PercentSalaryHike  StockOptionLevel  TotalWorkingYears  \
0                 1.0                 11                 0                1.0   
1                 0.0                 23                 1                6.0   
2                 1.0                 15                 3                5.0   
3                 3.0                 11                 3               13.0   
4                 4.0                 12                 2                9.0   

   TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
0                      6               1                        0   
1                      3               5                        1   
2                      2               5                        0   
3                      5               8                        7   
4                      2               6                        0   

   YearsWithCurrManager  
0                     0  
1                     4  
2                     3  
3                     5  
4                     4

### Split dataset

In [13]:
Y = df['Attrition']
X = df.drop('Attrition', axis=1)

X.head()

Age  BusinessTravel  Department  DistanceFromHome  Education  \
0   51               2           2                 6          2   
1   31               1           1                10          1   
2   32               1           1                17          4   
3   38               0           1                 2          5   
4   32               2           1                10          1   

   EducationField  Gender  JobLevel  JobRole  MaritalStatus  MonthlyIncome  \
0               1       0         1        0              1         131160   
1               1       0         1        6              2          41890   
2               4       1         4        7              1         193280   
3               1       1         3        1              1          83210   
4               3       1         1        7              2          23420   

   NumCompaniesWorked  PercentSalaryHike  StockOptionLevel  TotalWorkingYears  \
0                 1.0                 11                 0                1.0   
1                 0.0                 23                 1                6.0   
2                 1.0                 15                 3                5.0   
3                 3.0                 11                 3               13.0   
4                 4.0                 12                 2                9.0   

   TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
0                      6               1                        0   
1                      3               5                        1   
2                      2               5                        0   
3                      5               8                        7   
4                      2               6                        0   

   YearsWithCurrManager  
0                     0  
1                     4  
2                     3  
3                     5  
4                     4

### Train model

In [14]:
import statsmodels.api as sm

# Add constant to dataset
X1 = pd.DataFrame(
    sm.add_constant(X.values),
    columns=['const']+list(X.columns))
X1.head()

const   Age  BusinessTravel  Department  DistanceFromHome  Education  \
0    1.0  51.0             2.0         2.0               6.0        2.0   
1    1.0  31.0             1.0         1.0              10.0        1.0   
2    1.0  32.0             1.0         1.0              17.0        4.0   
3    1.0  38.0             0.0         1.0               2.0        5.0   
4    1.0  32.0             2.0         1.0              10.0        1.0   

   EducationField  Gender  JobLevel  JobRole  MaritalStatus  MonthlyIncome  \
0             1.0     0.0       1.0      0.0            1.0       131160.0   
1             1.0     0.0       1.0      6.0            2.0        41890.0   
2             4.0     1.0       4.0      7.0            1.0       193280.0   
3             1.0     1.0       3.0      1.0            1.0        83210.0   
4             3.0     1.0       1.0      7.0            2.0        23420.0   

   NumCompaniesWorked  PercentSalaryHike  StockOptionLevel  TotalWorkingYears  \
0                 1.0               11.0               0.0                1.0   
1                 0.0               23.0               1.0                6.0   
2                 1.0               15.0               3.0                5.0   
3                 3.0               11.0               3.0               13.0   
4                 4.0               12.0               2.0                9.0   

   TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
0                    6.0             1.0                      0.0   
1                    3.0             5.0                      1.0   
2                    2.0             5.0                      0.0   
3                    5.0             8.0                      7.0   
4                    2.0             6.0                      0.0   

   YearsWithCurrManager  
0                   0.0  
1                   4.0  
2                   3.0  
3                   5.0  
4                   4.0

In [15]:
# Train
model  = sm.Logit(Y.values.reshape(-1,1), X1)
result = model.fit()
result.summary()

Optimization terminated successfully.
         Current function value: 0.392916
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:                      y   No. Observations:                 4382
Model:                          Logit   Df Residuals:                     4362
Method:                           MLE   Df Model:                           19
Date:                Thu, 06 Aug 2020   Pseudo R-squ.:                  0.1093
Time:                        08:13:27   Log-Likelihood:                -1721.8
converged:                       True   LL-Null:                       -1933.1
Covariance Type:            nonrobust   LLR p-value:                 8.681e-78
===========================================================================================
                              coef    std err          z      P>|z|      [0.025      0.975]
-------------------------------------------------------------------------------------------
const                       0.0270      0.414      0.065      0.948      -0.785       0.839
Age                        -0.0307      0.007     -4.478      0.000      -0.044      -0.017
BusinessTravel             -0.0137      0.066     -0.209      0.834      -0.143       0.115
Department                 -0.2229      0.082     -2.735      0.006      -0.383      -0.063
DistanceFromHome           -0.0012      0.005     -0.231      0.818      -0.012       0.009
Education                  -0.0664      0.043     -1.555      0.120      -0.150       0.017
EducationField             -0.0954      0.034     -2.849      0.004      -0.161      -0.030
Gender                      0.0855      0.090      0.952      0.341      -0.091       0.262
JobLevel                   -0.0285      0.040     -0.716      0.474      -0.107       0.050
JobRole                     0.0400      0.018      2.226      0.026       0.005       0.075
MaritalStatus               0.5835      0.063      9.212      0.000       0.459       0.708
MonthlyIncome           -1.815e-06   9.57e-07     -1.897      0.058   -3.69e-06    6.01e-08
NumCompaniesWorked          0.1174      0.018      6.390      0.000       0.081       0.153
PercentSalaryHike           0.0126      0.012      1.067      0.286      -0.011       0.036
StockOptionLevel           -0.0675      0.052     -1.302      0.193      -0.169       0.034
TotalWorkingYears          -0.0584      0.012     -4.873      0.000      -0.082      -0.035
TrainingTimesLastYear      -0.1443      0.035     -4.097      0.000      -0.213      -0.075
YearsAtCompany              0.0132      0.018      0.718      0.473      -0.023       0.049
YearsSinceLastPromotion     0.1328      0.020      6.479      0.000       0.093       0.173
YearsWithCurrManager       -0.1394      0.022     -6.288      0.000      -0.183      -0.096
===========================================================================================
"""

The following features are not relevant to predict attrition:
* Business Traval
* Distance From Home
* Education
* Gender
* Job Level
* Percent Salary Hike
* Stock Option Level
* Year at Company

And we didn't need to add a constant.  
On the flip side, the relevant features are:
* Age
* Department
* Education Field
* Job Role
* Marital Status
* Monthly Income
* Num Companies Worked
* Total Working Years
* Years Since Last Promotion
* Years with Current Manager

## 3. Real Estate

### Import dataset

In [16]:
df = pd.read_excel('Linear Regression.xlsx', sheet_name=0)
df.head()

price  sqft_living  bedrooms  bathrooms  floors
0  221900         1180         3       1.00     1.0
1  538000         2570         3       2.25     2.0
2  180000          770         2       1.00     1.0
3  604000         1960         4       3.00     1.0
4  510000         1680         3       2.00     1.0

### Handle missing values

In [17]:
df.isna().sum()

price          0
sqft_living    0
bedrooms       0
bathrooms      0
floors         0
dtype: int64

### Split dataset

In [18]:
Y = df['price']
X = df.drop('price', axis=1)

X.head()

sqft_living  bedrooms  bathrooms  floors
0         1180         3       1.00     1.0
1         2570         3       2.25     2.0
2          770         2       1.00     1.0
3         1960         4       3.00     1.0
4         1680         3       2.00     1.0

### Train model 1: {price/sqft_living}

In [19]:
import statsmodels.api as sm

lm = sm.OLS(Y, X['sqft_living'])
model = lm.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  price   R-squared (uncentered):                   0.839
Model:                            OLS   Adj. R-squared (uncentered):              0.839
Method:                 Least Squares   F-statistic:                          1.126e+05
Date:                Thu, 06 Aug 2020   Prob (F-statistic):                        0.00
Time:                        08:13:28   Log-Likelihood:                     -3.0032e+05
No. Observations:               21613   AIC:                                  6.006e+05
Df Residuals:                   21612   BIC:                                  6.006e+05
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
sqft_living   263.0892      0.784    335.597      0.000     261.553     264.626
==============================================================================
Omnibus:                    16043.334   Durbin-Watson:                   1.980
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           692411.844
Skew:                           3.130   Prob(JB):                         0.00
Kurtosis:                      30.013   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The size of the living room is a relevant feature to predict the price (p-value < 0.05).  
83.9% of the variation in price is explained by this feature.

### Train model 2: {price/bedrooms}

In [20]:
lm = sm.OLS(Y, X['bedrooms'])
model = lm.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  price   R-squared (uncentered):                   0.711
Model:                            OLS   Adj. R-squared (uncentered):              0.711
Method:                 Least Squares   F-statistic:                          5.323e+04
Date:                Thu, 06 Aug 2020   Prob (F-statistic):                        0.00
Time:                        08:13:28   Log-Likelihood:                     -3.0663e+05
No. Observations:               21613   AIC:                                  6.133e+05
Df Residuals:                   21612   BIC:                                  6.133e+05
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
bedrooms    1.575e+05    682.663    230.713      0.000    1.56e+05    1.59e+05
==============================================================================
Omnibus:                    17896.494   Durbin-Watson:                   1.958
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          1088783.142
Skew:                           3.594   Prob(JB):                         0.00
Kurtosis:                      37.020   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The number of bedrooms is a relevant feature to predict the price (p-value < 0.05).  
71.1% of the variation in price is explained by this feature.

### Train model 3: {price/bathrooms}

In [21]:
lm = sm.OLS(Y, X['bathrooms'])
model = lm.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  price   R-squared (uncentered):                   0.771
Model:                            OLS   Adj. R-squared (uncentered):              0.771
Method:                 Least Squares   F-statistic:                          7.280e+04
Date:                Thu, 06 Aug 2020   Prob (F-statistic):                        0.00
Time:                        08:13:28   Log-Likelihood:                     -3.0412e+05
No. Observations:               21613   AIC:                                  6.082e+05
Df Residuals:                   21612   BIC:                                  6.082e+05
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
bathrooms   2.548e+05    944.330    269.818      0.000    2.53e+05    2.57e+05
==============================================================================
Omnibus:                    17130.465   Durbin-Watson:                   1.960
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           861569.543
Skew:                           3.416   Prob(JB):                         0.00
Kurtosis:                      33.167   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The number of bathrooms is a relevant feature to predict the price (p-value < 0.05).  
77.1% of the variation in price is explained by this feature.

### Train model 4: {price/floors}

In [22]:
lm = sm.OLS(Y, X['floors'])
model = lm.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  price   R-squared (uncentered):                   0.684
Model:                            OLS   Adj. R-squared (uncentered):              0.684
Method:                 Least Squares   F-statistic:                          4.672e+04
Date:                Thu, 06 Aug 2020   Prob (F-statistic):                        0.00
Time:                        08:13:29   Log-Likelihood:                     -3.0761e+05
No. Observations:               21613   AIC:                                  6.152e+05
Df Residuals:                   21612   BIC:                                  6.152e+05
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
floors      3.399e+05   1572.369    216.139      0.000    3.37e+05    3.43e+05
==============================================================================
Omnibus:                    16926.427   Durbin-Watson:                   1.932
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           841619.223
Skew:                           3.354   Prob(JB):                         0.00
Kurtosis:                      32.826   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The number of floors is a relevant feature to predict the price (p-value < 0.05).  
68.4% of the variation in price is explained by this feature.

### Train model 5: {price/sqft_living,bedrooms,bathrooms,floors}

In [23]:
lm = sm.OLS(Y, X)
model = lm.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  price   R-squared (uncentered):                   0.843
Model:                            OLS   Adj. R-squared (uncentered):              0.843
Method:                 Least Squares   F-statistic:                          2.912e+04
Date:                Thu, 06 Aug 2020   Prob (F-statistic):                        0.00
Time:                        08:13:29   Log-Likelihood:                     -3.0001e+05
No. Observations:               21613   AIC:                                  6.000e+05
Df Residuals:                   21609   BIC:                                  6.001e+05
Df Model:                           4                                                  
Covariance Type:            nonrobust                                                  
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
sqft_living   305.3462      3.065     99.613      0.000     299.338     311.354
bedrooms    -4.318e+04   1802.601    -23.956      0.000   -4.67e+04   -3.96e+04
bathrooms    1.064e+04   3811.694      2.791      0.005    3165.360    1.81e+04
floors       1.618e+04   3406.591      4.749      0.000    9499.847    2.29e+04
==============================================================================
Omnibus:                    13983.158   Durbin-Watson:                   1.983
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           457060.878
Skew:                           2.622   Prob(JB):                         0.00
Kurtosis:                      24.910   Cond. No.                     5.70e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.7e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

All features are relevant to predict the price (p-value < 0.05).  
In total, 84.3% of the variation in price is explained by the features.